packages/@uppy/provider-views/src/ProviderView/ProviderView.js

```diff
@@ -226,9 +226,14 @@ module.exports = class ProviderView extends View {
     const link = this.provider.authUrl({ state: authState, uppyVersions: clientVersion })
 
     const authWindow = window.open(link, '_blank')
+
     const handleToken = (e) => {
-      if (!this.#isOriginAllowed(e.origin, this.plugin.opts.companionAllowedHosts) || e.source !== authWindow) {
-        this.plugin.uppy.log(`rejecting event from ${e.origin} vs allowed pattern ${this.plugin.opts.companionAllowedHosts}`)
+      if (e.source !== authWindow) {
+        this.plugin.uppy.log('rejecting event from unknown source')
+        return
+      }
+      if (!this.#isOriginAllowed(e.origin, this.plugin.opts.companionAllowedHosts)) {
+        this.plugin.uppy.log(`rejecting event from ${e.origin} vs allowed pattern ${this.plugin.opts.companionAllowedHosts}`, 'error')
```

Comment:
wouldn't they be caught in the above if-statement `if (e.source !== authWindow)`? because they would come from a different `event.source`

Reply:
oh yeah! So this would only happen if the auth window went into some other website somehow or if the configuration is wrong. then I agree 'error' is right. :)